src/states/joining_peer.rs

```diff
@@ -47,52 +51,67 @@ pub struct JoiningPeerDetails {
     pub network_cfg: NetworkParams,
     pub timer: Timer,
     pub rng: MainRng,
-    pub elders_info: EldersInfo,
-    pub relocate_payload: Option<RelocatePayload>,
 }
 
 // State of a node after bootstrapping, while joining a section
 pub struct JoiningPeer {
+    stage: Stage,
```

Comment:
Yes, but that's the point. We are inverting the pattern we use - instead of having the top level `StateMachine` type which is just responsible for switching the states and the actual logic is implemented in the states, we will have one top level type where all the logic will be and it will contain this `Stage` enum which just contains the stuff that's different between the stages (I call them stages now instead of states which seems more appropriate).